lib/cylc/tests/test_config.py

```diff
@@ -0,0 +1,117 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+import tempfile
+
+from cylc.config import SuiteConfig
+
+
+def get_test_inheritance_quotes():
+    """Provide test data for test_family_inheritance_and_quotes."""
+    return [
+        # first case, second family name surrounded by double quotes
+        b'''
+[cylc]
+    [[parameters]]
+        major = 1..5
+        minor = 10..20
+[scheduling]
+    [[dependencies]]
+        graph = """hello => MAINFAM<major, minor>
+                   hello => SOMEFAM
+        """
+[runtime]
+    [[root]]
+        script = true
+    [[MAINFAM<major, minor>]]
+    [[SOMEFAM]]
+    [[ goodbye_0<major, minor> ]]
+        inherit = 'MAINFAM<major, minor>', "SOMEFAM"
+        ''',
+        # second case, second family surrounded by single quote
+        b'''
+[cylc]
+    [[parameters]]
+        major = 1..5
+        minor = 10..20
+[scheduling]
+    [[dependencies]]
+        graph = """hello => MAINFAM<major, minor>
+                   hello => SOMEFAM
+        """
+[runtime]
+    [[root]]
+        script = true
+    [[MAINFAM<major, minor>]]
+    [[SOMEFAM]]
+    [[ goodbye_0<major, minor> ]]
+        inherit = 'MAINFAM<major, minor>', 'SOMEFAM'
+        ''',
+        # third case, second family name without quotes
+        b'''
+[cylc]
+    [[parameters]]
+        major = 1..5
+        minor = 10..20
+[scheduling]
+    [[dependencies]]
+        graph = """hello => MAINFAM<major, minor>
+                   hello => SOMEFAM
+        """
+[runtime]
+    [[root]]
+        script = true
+    [[MAINFAM<major, minor>]]
+    [[SOMEFAM]]
+    [[ goodbye_0<major, minor> ]]
+        inherit = 'MAINFAM<major, minor>', SOMEFAM
+        '''
+    ]
+
+
+class TestConfig(unittest.TestCase):
+
+    def test_family_inheritance_and_quotes(self):
+        """Test that inheritance does not ignore items, if not all quoted.
+
+        For example:
+
+            inherit = 'MAINFAM<major, minor>', SOMEFAM
+            inherit = 'BIGFAM', SOMEFAM
+
+        See bug #2700 for more/
+        """
+        template_vars = {}
+        for content in get_test_inheritance_quotes():
+            with tempfile.NamedTemporaryFile() as tf:
+                tf.write(content)
+                tf.flush()
+                config = SuiteConfig(
+                    'test',
+                    tf.name,
+                    template_vars=template_vars)
+                self.assertTrue('goodbye_0_major1_minor10'
+                                in config.runtime['descendants']
+                                ['MAINFAM_major1_minor10'])
+                self.assertTrue('goodbye_0_major1_minor10'
+                                in config.runtime['descendants']['SOMEFAM'])
```

Comment:
Can use `.assertIn`?

Reply:
:open_mouth: ! Didn't know about it, updating the code in a few minutes.

Reply:
Also could just do it the pytest way.

Reply:
I wondered if we should start transitioning the tests to pytest... they offer test parameters too (can't recall the syntax now tho)

Reply:
:+1: for pytest, but we probably should defer to a separate PR - with a strategy to move most unit-like tests to pytest.

Reply:
We can leave our `unittest` based unit tests as they are, just write new ones with pytest magic.